File: Sean的leetcode/Leetcode/Flip Game.cpp
class Solution {
public:
    vector<string> generatePossibleNextMoves(string s) {
        vector<string> res;
        int n = s.size();
        if (n == 0) return res;
        for (int i = 0; i < n - 1; i++)
        {
            if (s[i] == '+' && s[i+1] == '+')
                res.push_back(s.substr(0,i)+"--"+s.substr(i+2));
        }
        return res;
    }
};

File: Sean的leetcode/python/Flip Game.py
###
# String
# Time Complexity: O(n)
# Space Complexity: O(n^2)
###
class Solution(object):
    def generatePossibleNextMoves(self, s):
        """
        :type s: str
        :rtype: List[str]
        """
        if len(s) <= 1:
            return []
        re = []
        for i in xrange(len(s) - 1):
            if s[i] == "+" and s[i+1] == "+":
                re.append(s[:i] + "--" + s[i+2:])       
        return re
        
        